### PR TITLE
Native plot

### PR DESCRIPTION
**Pull request by ltindall**

### CODE REVIEW

File: sigscan.py [hunk 1]
```diff
@@ -220,10 +220,11 @@ def main(config=None, debug=False):
         planner = MCTSPlanner(env, actions, depth, c, simulations)
 
     # Flask
-    fig = Figure(figsize=(10,10))
+    fig = plt.figure(figsize=(10,10))
     ax = fig.subplots()
     time_step = 0
-    run_flask(flask_host, flask_port, fig, results, debug)
+    if config.get('flask', False) in [True, 'True', 'true']:
```
Comment (cglewis) [inline sigscan.py:226]: `False` should be of type str `"False"`

File: sigscan.py [hunk 2]
```diff
@@ -247,9 +248,12 @@ def main(config=None, debug=False):
 
         plot_start = timer()
         results.live_plot(env=env, time_step=time_step, fig=fig, ax=ax, data=data, simulated=False, textstr=textstr)
-        plot_end = timer()
-
-        particle_save_start = timer()
+        if config.get('native_plot', False):
```
Comment (cglewis) [inline sigscan.py:251]: `False` should be of type str `"False"`

**Review by cglewis — CHANGES_REQUESTED** [review]

**Comment by cglewis** [comment]
@ltindall looks like matplotlib needs to be reworked a bit more:
```
sigscan.py:253: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.
```

**Comment by ltindall** [comment]
From [here](https://stackoverflow.com/questions/56656777/userwarning-matplotlib-is-currently-using-agg-which-is-a-non-gui-backend-so) could you try 
`sudo apt-get install python3-tk`

**Comment by cglewis** [comment]
@ltindall these changes fix the matplotlib warning:
```
diff --git a/Dockerfile b/Dockerfile
index cd22fd5..84c18d3 100644
--- a/Dockerfile
+++ b/Dockerfile
@@ -1,8 +1,9 @@
 FROM nvidia/cuda:11.7.0-base-ubuntu20.04
 LABEL maintainer="Lucas Tindall <ltindall@iqt.org>"
+ENV DEBIAN_FRONTEND=noninteractive
 ENV PYTHONUNBUFFERED 1
 
-RUN apt-get update && apt-get install -y libjpeg-dev python3 python3-pip
+RUN apt-get update && apt-get install -y --no-install-recommends libjpeg-dev python3 python3-pip python3-tk
 RUN python3 -m pip install --upgrade pip
 
 COPY requirements.txt requirements.txt
```

**Review by cglewis — APPROVED** [review]

**Pull request merged**